helix-term/src/commands.rs

```diff
@@ -4641,6 +4650,14 @@ fn join_selections_impl(cx: &mut Context, select_space: bool) {
             let start = line_end_char_index(&slice, line);
             let mut end = text.line_to_char(line + 1);
             end = skip_while(slice, end, |ch| matches!(ch, ' ' | '\t')).unwrap_or(end);
+            let slice_from_end = slice.slice(end..);
+            for pfx in comment_prefixes.iter() {
+                if slice_from_end.starts_with(pfx) {
+                    end += pfx.len();
```

Comment:
This is mixing up byte and character indexing: `end` is a character index into the rope and the token's `len()` is the number of bytes in its UTF-8 representation. It won't make a difference in the common case (comment token is ASCII) but if you try a comment token that takes more than one byte in UTF-8 `end` will be greater than it should. Instead here we have to use `token.chars().count()`.